assemblies/resources/bin/check_ports

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+# Licensed to The Apereo Foundation under one or more contributor license
+# agreements. See the NOTICE file distributed with this work for additional
+# information regarding copyright ownership.
+#
+# The Apereo Foundation licenses this file to you under the Educational
+# Community License, Version 2.0 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of the License
+# at:
+#
+#   http://opensource.org/licenses/ecl2.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# -----------------------------------------------------------------------
+# check if http or https port is available to run opencast otherwise exit
+
+PROGNAME=$(basename "$0")
+
+if [ "$PROGNAME" = "start-opencast" ]; then
+
+  if [ -d "$1" ]; then
+    SETENV_ETC="$1"
+  else
+    if [ -d "${KARAF_BASE}" ]; then
+      SETENV_ETC="${KARAF_BASE}/etc"
+    else
+      SETENV_ETC="etc"
+    fi
+  fi
+
+  check_port() {
+    a="$1"
+    if [ "$a" -eq "$a" ] 2>/dev/null ; then
+      port="$a"
+      hexPort=$(printf ":%04X " "$a")
+    else
+      b="${a##*:}"
+      port=$(eval echo "\$${b%%\}*}")
+      if [ "$port" -eq "$port" ] 2>/dev/null ; then
+        hexPort=$(printf ":%04X " "$port")
+      else
+        return 0;
+      fi
+    fi
+    if [ -d /proc/net ] ; then
+      for i in /proc/net/*; do
+        if  [ -f "$i" ] && [ -r "$i" ]; then
+          o=$(grep -c "$hexPort" "$i")
+          if [ "$o" = "1" ]; then
+            return 1;
+          fi
+        fi
+      done
+    else
+        if [ -x "$(command -v netstat)" ]; then
+          o=$(netstat -anp tcp | grep -i "listen" | grep -c -F ".$port ")
+          if [ "$o" -gt 0 ] ; then
+            return 1;
+          fi
+        else
+          return 0;
+        fi
+    fi
+    return 0;
+  }
+
+  usehttp=$(grep "org.osgi.service.http.enabled[ =]" "${SETENV_ETC}"/org.ops4j.pax.web.cfg)
+  if [ "${usehttp##*=}" = "true" ]; then
+    httpport=$(grep "org.osgi.service.http.port[ =]" "${SETENV_ETC}"/org.ops4j.pax.web.cfg)
+    check_port "${httpport##*=}"
+    if [ "$?" = "1" ]; then
+      printf "\nHttp server port \e[1m%s\e[0m already in use \n" "${httpport##*=}"
+      printf "reconfigure \e[1morg.osgi.service.http.port\e[0m in \e[1m%s/org.ops4j.pax.web.cfg\e[0m \n" "${SETENV_ETC}"
+      exit 1
+    fi
+  fi
+
+  usehttps=$(grep "org.osgi.service.http.secure.enabled[ =]" ${SETENV_ETC}/org.ops4j.pax.web.cfg)
+  if [ "${usehttps##*=}" = "true" ]; then
```

Comment:
The Pax Web interpretation of this value is case insensitive. Hence, it could also be `TrUe` and there could be spaces after the `=`. Maybe something like:

```sh
if grep -qi '^org.osgi.service.http.secure.enabled[ =]*true *$' "${SETENV_ETC}/org.ops4j.pax.web.cfg"; then
…
```

Reply:
fixed